### PR TITLE
v1.6: Proximity precision index setting

### LINKED ISSUE #2653: v1.6: Proximity setting
v1.6 introduces a new index setting, `proximityPrecision`. It accepts one of two string values: `wordScale` and `attributeScale`. `wordScale` is the default and presents no significant change from previous Meilisearch releases. `attributeScale` makes calculations simpler and faster, but less precise: instead of counting the distance between terms, Meilisearch only checks whether search terms are present on the same attribute. Disabling this ranking rule might impact relevancy.

## Tasks
- [x] Update `/reference/api/settings.mdx`
- [x] Update `/learn/core_concepts/relevancy.mdx`

## References
- Usage: https://www.notion.so/meilisearch/Customize-proximity-precision-usage-aa69c2bab2c3402bab9340ae4def4577
- engine issue: https://github.com/meilisearch/meilisearch/issues/4187
- SME: @ManyTheFish

### PR DESCRIPTION
**Pull request by guimachiavelli**
Closes #2653

### CODE REVIEW

**Comment by guimachiavelli** [comment]
Tagging @mdubus since we spoke about this new setting last week. No action needed, just in case there's anything relevant for you here. 👍

File: reference/api/settings.mdx [hunk 1]
```diff
@@ -886,6 +889,103 @@ Reset an index's pagination settings to their [default value](#pagination-object
 
 You can use the returned `taskUid` to get more details on [the status of the task](/reference/api/tasks#get-one-task).
 
+## Proximity precision
+
+Calculating the [proximity ranking rule](/learn/core_concepts/relevancy#3-proximity) is a resource-intensive operation. Lowering its precision may significantly improve performance and will have little impact on result relevancy in most use-cases.
```
[the remark this answers is not in the document]
Comment (ManyTheFish) [reply reference/api/settings.mdx:894]: An additional point: this change doesn't impact the proximity ranking rule only. The phrase search feature is impacted similarly; Meilisearch will no longer ensure that the words of a phrase search are close together and in good order but only ensure they are in the same attribute. ☺️

**Review by curquiza — APPROVED** [review]
ok for code samples

**Review by ManyTheFish — APPROVED** [review]
LGTM

File: reference/api/settings.mdx [hunk 2]
```diff
@@ -886,6 +889,103 @@ Reset an index's pagination settings to their [default value](#pagination-object
...
+
+#### Example
+
+<CodeSamples id="get_proximity_precision_settings_1" />
+
+##### Response: `200 OK`
+
+```json
+{
+  "proximityPrecision": "byWord"
```
Comment (honhimW) [inline reference/api/settings.mdx:921]: GET /indexes/{index_uid}/settings/proximity-precision does not return a `JSON object` but a `JSON string`

#### Response: `200 OK`

```json
"byWord"|"byAttribute"
```

Comment (curquiza) [reply reference/api/settings.mdx:921]: I opened https://github.com/meilisearch/documentation/issues/2693
thank you for the feedback

**Pull request merged**